platform/android/app/src/main/kotlin/coop/polypoly/polypod/ErrorUploader.kt

```diff
@@ -0,0 +1,33 @@
+package coop.polypoly.polypod
+import android.content.Context
+import coop.polypoly.polypod.endpoint.Endpoint
+import coop.polypoly.polypod.logging.LoggerFactory
+
+object ErrorUploader {
+    private val logger = LoggerFactory.getLogger(javaClass.enclosingClass)
+
+    suspend fun uploadToServer(context: Context, errorMsg: String) {
+        val endpointId = "polyApiErrorReport"
+        val endpoint = Endpoint(context)
+
+        val endpointInfo = endpoint.endpointInfofromId(endpointId)
+        if (endpointInfo == null) {
+            logger.error(
+                "uploadToServer: No endpoint found under that endpointId"
+            )
+            throw PodApiError().endpointError()
+        }
```

Comment:
AFAIU as I understand this is needed to extract the authorisation for a given endpoint, it could have been better if this is part of the Endpoint itself, then we could just use the Endpoint to upload the error, without the need of creating ErrorUploader wrapper.